meson.build

```diff
@@ -303,69 +299,68 @@ if not headers_only
 		endif
 	endif
 
-	if not static
-		ldso_lib = shared_library('ld', rtdl_sources,
-				name_prefix: '',
-				cpp_args: ['-fvisibility=hidden', '-DMLIBC_BUILDING_RTDL',
-					'-fno-stack-protector'],
-				include_directories: rtdl_include_dirs,
-				dependencies: rtdl_deps + rtlib_deps,
-				install: true)
-
-		libc = shared_library('c',
-				[
-					libc_sources,
-					internal_sources,
-					ansi_sources,
-					lsb_sources,
-				],
-				include_directories: libc_include_dirs,
-				dependencies: libc_deps + rtlib_deps,
-				link_with: [ldso_lib],
-				link_whole: libc_sublibs,
-				install: true)
-
-		shared_library('dl', 'dummy-libs/libdl/src/dummy.cpp', install: true)
-		shared_library('pthread', 'dummy-libs/libpthread/src/dummy.cpp', install: true)
-		shared_library('rt', 'dummy-libs/librt/src/dummy.cpp', install: true)
-		shared_library('util', 'dummy-libs/libutil/src/dummy.cpp', install: true)
-		shared_library('m', 'dummy-libs/libm/src/dummy.cpp', install: true)
-		if not disable_crypt_option
-			shared_library('crypt', 'dummy-libs/libcrypt/src/dummy.cpp', install: true)
-		endif
-		shared_library('resolv', 'dummy-libs/libresolv/src/dummy.cpp', install: true)
-	else
-		ldso_lib = static_library('ld', rtdl_sources,
-				name_prefix: '',
-				cpp_args: ['-fvisibility=hidden', '-DMLIBC_BUILDING_RTDL', '-DFRIGG_HAVE_LIBC',
-				'-fno-stack-protector'],
-				include_directories: rtdl_include_dirs,
-				dependencies: rtdl_deps,
-				install: false)
-		libc = static_library('c',
-				[
-					libc_sources,
-					internal_sources,
-					ansi_sources,
-					lsb_sources,
-				],
-				cpp_args: ['-DFRIGG_HAVE_LIBC', '-fno-stack-protector'],
-				c_args: ['-fno-stack-protector'],
-				include_directories: libc_include_dirs,
-				dependencies: libc_deps,
-				link_with: [ldso_lib],
-				link_whole: [libc_sublibs, ldso_lib],
-				install: true)
-
-		static_library('pthread', 'dummy-libs/libpthread/src/dummy.cpp', install: true)
-		static_library('rt', 'dummy-libs/librt/src/dummy.cpp', install: true)
-		static_library('util', 'dummy-libs/libutil/src/dummy.cpp', install: true)
-		static_library('m', 'dummy-libs/libm/src/dummy.cpp', install: true)
-		if not disable_crypt_option
-			static_library('crypt', 'dummy-libs/libcrypt/src/dummy.cpp', install: true)
-		endif
-		static_library('resolv', 'dummy-libs/libresolv/src/dummy.cpp', install: true)
+	ld_cpp_args = [
+		'-fvisibility=hidden',
+		'-fno-stack-protector',
+		'-DMLIBC_BUILDING_RTDL'
+	]
+
+	libc_all_sources = [
+		libc_sources,
+		internal_sources,
+		ansi_sources,
+		lsb_sources,
+	]
+
+	# Our library have different behaviour when built as static and shared libraries.
+	# Hence we need to rebuild the object files with a different define for each mode.
+	if library_type in ['static', 'both']
+		static_cpp_args = [
+			'-DMLIBC_STATIC_BUILD',
+			'-DFRIGG_HAVE_LIBC',
```

Comment:
Is this define even still used? It seems weird to me that it's defined for the static case but not the dynamic case.